src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -129,22 +157,73 @@ def createPileup(self, pdict, rseList):
             return [err.error()]
         return []
 
-    def updatePileup(self, doc, rseList=None, validate=True):
+    def updatePileup(self, doc, rseList=None, validate=True, userDN=''):
         """
         Update pileup data with provided input
 
         :param doc: a dictionary of pieleup data to be updated
         :param rseList: a list of valid RSE names
         :param validate: boolean defining whether the doc needs
                          to be validated or not
+        :param userDN: a string, user DN used in HTTP PUT request
         :return: list of MSPileupError or empty list
         """
+        self.logger.info("Updating pileup document %s", doc)
         rseList = rseList or []
+
+        # what logic should we use to lookup document when we are given
```

Comment:
I only asked since in original discussion and implementation we spoke that either pileup name or custom name can be provided. Now, we outlined logic which requires only pileup name in API call.